src/components/textToSpeech/js/TextToSpeech.js

```diff
@@ -170,7 +242,11 @@ var fluid_2_0_0 = fluid_2_0_0 || {};
             paused: false
         };
 
-        if (!that.queue.length) {
+        if (that.queue.currentUtterance) {
+            that.queue.currentUtterance = undefined;
```

Comment:
Agree this is better; I had to refactor the queue behaviour a little so `that.queue.shift` happens at `handleEnd` rather than `handleStart` or we run into the same premature garbage collection behaviour in Safari that we were seeing previously. I don't think this is a problematic change (it also lets `handleStart` simply be a changePath record rather than a function), but please let me know if you foresee an issue with that behavioural change.